src/ansys/units/quantity.py

```diff
@@ -15,7 +15,7 @@
 
 class Quantity:
     """
-    A class which contains a physical quantity's value and associated units.
+    A class containing a physical quantity's value and associated units.
```

Comment:
"representing" might be a better alternative.

Can also remove use of future tense (will only work) below.

NumPy wording can be more precise: "A ``Quantity`` object can be instantiated from a ``NumPy`` array or list only if the NumPy package is installed."

Re "A Quantity instance will contain both SI units and the SI value to
    facilitate consistent computation with other quantities." Is this an implementation detail or is it a significant comment about the obervable behaviour? The comment is unclear.

This whole soc section should be split up into separate paragraphs. Some bits already are but some of the rest is a big block of unconnected statements run together. It doesn't read well.

Reply:
TODO apply these. Also extend user guide regarding the float conversion behaviour.